pkg/edvprovider/edvprovider.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package edvprovider
+
+import (
+	"github.com/trustbloc/edv/pkg/restapi/edv/models"
+
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+// EDVProvider represents a provider with functionality needed for EDV data storage.
```

Comment:
@fqutishat @bstasyszyn Take a look at my strategy here and let me know what you think.

The reason for this "EDVProvider" is that I need to do that CouchDB document index workaround (described in the comments in CouchDBEDVProvider). That workaround is specific to EDV, which is why I did not do it in edge-core.

I really wasn't sure what to name these or whether or not this was the best way to do it. I'm aware there are likely issues here... so let me know if you can think of improvements :)

Reply:
This comment is out of date now...disregard it.